File: codechef/PPOW.java
import java.util.*;
import java.lang.*;
import java.io.*;
import java.io.InputStream;
import java.io.InputStreamReader;

/* Name of the class has to be "Main" only if the class is public. */
class PPOW
{
	public long sumOfDigits(int base, int exp) {
		if(base==1 || exp==0)
			return 1;
		int numberOfDigits = (int) Math.ceil(exp * Math.log10(base));
		int[] digits = new int[numberOfDigits];
		digits[0] = base;
		int currentExp = 1;
		while (currentExp < exp) {
			currentExp++;
			int carry = 0;
			for (int i = 0; i < digits.length; i++) {
				int num = base * digits[i] + carry;
				digits[i] = num % 10;
				carry = num / 10;
			}
		}
		long sum = 0;
		for (int digit : digits)
		sum += digit;
		return sum;
	}
	public static void main (String[] args) throws java.lang.Exception
	{
		Scanner in = new Scanner(System.in);
		int num = in.nextInt();
		int iter=1;
		while(num!=0){
			num=num-1;
			int base = in.nextInt();
			int exp = in.nextInt();
			//int base = 9;
			//int exp = 4000;
			System.out.println("Case "+iter+": "+new PPOW().sumOfDigits(base, exp));
			iter++;
		}
	}
}

File: codechef/PPOW.cpp
#include<iostream>
#include<string>
#include<cstring>
#include<vector>
#include<algorithm>
#include<cmath>
using namespace std;
int n,rem,sum;
int main(){
	long int t;
	int a,b,iter=1;;
	cin >> t;
	while(t--){
		cin >> a >> b;
		vector<int> digits(50000);
		digits[0]=a;
		digits[1]=a;
		for(int k=2;k<b+1;k++){
			for(int i=1;i<=digits[0];i++){
				digits[i]=digits[i]*a+rem;
				rem=0;
				if(digits[i]>9){
					rem=digits[i]/10;
					digits[i]%=10;
					if(i==digits[0])
						digits[0]++;
				}
			}
		}
		for(int i=digits[0];i>=1;i--){
			sum=sum+digits[i];
			//cout << digits[i];
		}
		cout << "Case " << iter++ << ": " << sum << endl;
		rem=0;
		sum=0;
	}
	return 0;
}
